plugin/code_actions.py

```diff
@@ -277,10 +277,9 @@ def run(self, edit: sublime.Edit, event: Optional[dict] = None,
             only_kinds: Optional[List[str]] = None, commands_by_config: CodeActionsByConfigName = None) -> None:
         self.commands = []  # type: List[Tuple[str, str, CodeActionOrCommand]]
         self.commands_by_config = {}  # type: CodeActionsByConfigName
-        self.has_commands_by_config = False
         if commands_by_config:
-            self.has_commands_by_config = True
-            self.handle_responses_async(commands_by_config)
+            self.commands_by_config = commands_by_config
```

Comment:
> That's not what I meant. This is unnecessarily stateful. I meant to pass the `has_commands_by_config` flag to `handle_responses_async`.

Well, it seems more streamlined, To be honest I don't like to pass more parameters

Reply:
But storing properties on the instance of TextCommand is a bad practice since it's initialized per-view rather than per-execution. So then there needs to be an extra code that resets the properties on run.